code/controllers/subsystem/input.dm

```diff
@@ -29,12 +42,53 @@ SUBSYSTEM_DEF(input)
 	)
 
 // Badmins just wanna have fun ♪
-/datum/controller/subsystem/input/proc/refresh_client_macro_sets()
+/datum/controller/subsystem/verb_manager/input/proc/refresh_client_macro_sets()
 	var/list/clients = GLOB.clients
 	for(var/i in 1 to clients.len)
 		var/client/user = clients[i]
 		user.set_macros()
 
-/datum/controller/subsystem/input/fire()
-	for(var/mob/user as anything in GLOB.keyloop_list)
-		user.focus?.keyLoop(user.client)
+/datum/controller/subsystem/verb_manager/input/can_queue_verb(datum/callback/verb_callback/incoming_callback, control)
+	//make sure the incoming verb is actually something we specifically want to handle
+	if(control != "mapwindow.map")
+		return FALSE
```

Comment:
why?

Reply:
code assumes the existence of a clicked atom which is only when the click happened in the map window.